terminal: 'bufhidden', 'modified'

justinmk opened the pull request:
- default to bufhidden=hide
- implement 'modified'

justinmk committed:
terminal: Set 'bufhidden=hide' and 'modified' by default.

- 'bufhidden=hide' allows intuitive behavior even with global 'nohidden'.
- 'modified' is a natural way to prevent loss of work. And the user can
  set 'nomodified' for cases where they don't want to be prompted.

- With bufhidden={delete,wipe}: terminal buffer is restarted when the
  user returns to a deleted/wiped terminal buffer.

Closes #2368
Closes #3340
References #5431

Remove some special-case checks; instead leverage existing checks for
'bufhidden' (b_p_bh) and 'buftype' (b_p_bt).

TODO:
- With bufhidden=unload, the buffer is blank if the user returns to an
  unloaded terminal buffer. We should instead save the terminal
  scrollback to a file, then reload that.

justinmk committed:
terminal: Set 'modified' by default.

This is important for:
- Consistency: previously we asked to confirm :bdelete, but not :qall.
  Also now that 'bufhidden' is respected, 'modified' is more meaningful.
- Flexibility: Users can set 'nomodified' if they never want to be asked
  to confirmed closing a :terminal buffer.
- Concepts: e.g. TextChangedT event implies that the text was changed.

Special case: We intentionally don't show "Modified" in the 'statusline'
nor 'tabline', because :terminal buffers are never *written*, it would
just be obnoxious to always show "Modified" flag.

jamessan commented on runtime/doc/vim_diff.txt:
```diff
@@ -147,13 +147,17 @@ Highlight groups:
+certain features removed/added at compile-time.  This is like if Vim's "H
+parent E"
+build was the only Vim release type (except Nvim is smaller than Vim's "H
+parent E"
```
For some reason, every `ug` you typed in this section was replaced with `\nparent `.

justinmk replied:
Oops, that's my bank password.

justinmk commented on runtime/doc/os_win32.txt:
```diff
@@ -1,206 +0,0 @@
-
-The directory of the Vim executable is appended to $PATH.  This is mostly to
-make "!xxd" work, as it is in the Tools menu.  And it also means that when
-executable() returns 1 the executable can actually be executed.
```
This is already covered at `:h executable()`.

justinmk commented on runtime/doc/os_win32.txt:
```diff
@@ -1,206 +0,0 @@
-	:set shellpipe=\|\ tee
-<  to your _vimrc.
-
-							*:!start*
```
Moved to `various.txt`, near `:h :!`.

justinmk committed:
test: terminal 'bufhidden'

DarkDefender committed:
tui: fix truecolor in libvte, tmux #7037

Closes #7041

justinmk committed:
doc: tui colors, cursor

justinmk committed:
options: remove ghost of `:set termcap` (#7102)

Closes #6763

justinmk committed:
vim-patch:8.0.0498 (#7103)

Problem:    Two autocmd tests are skipped on MS-Windows.
Solution:   Make the test pass on MS-Windows. Write the messages in a file
            instead of getting the output of system().
https://github.com/vim/vim/commit/e94260f3582cf928cc59e792b3b6ede57794a4a6

cloudhead commented:
Hey, any idea when this will land? Would love to have this. In the meantime is there anything I can do in my vim config to stop terminal buffers from being reset when their window is closed?

justinmk commented:
Yep, this is important to me too. Hopefully october.

plexigras commented:
so whats the status on this?

diego898 commented:
just checking on this!

igrep commented:
I'm looking forward to this change is merged!
